### DISCUSSION TITLE
Support multiple floors

### DISCUSSION THREAD

**Discussion by carefulcomputer** [sca075/mqtt_vacuum_camera#90] [Ideas]
### Description

I use https://github.com/pkoehlers/maploader to switch floor and maps. Map switcher works great. When I change the floor then the this map gets weird since the original segment lines stay but map has changed underneath it.

### Solution

Allow switching the map. Or if possible get the segments from the vacuum.

**Comment by sca075** [comment]
@carefulcomputer I'm not familiar with this vacuum add on you use.

Technically this camera already provides the segments the vacuum provides (segments = rooms). The segments coordinates are calculated based on the vacuum json data provided in the map_data topic.

While this plugin creates a backup of the map and restore it automatically when you change the "floor" (I understand Valetudo will restart to load the new map data each time).. so a new floor will be displayed (if you are running v1.5.9-rc2 the status text you might show that the vacuum is disconnected and stays frozen to the previous floor) until start/stop as suggested in the link you provided. In this case you would need to restart the camera too to update calibration and floor data.

So let's draft the specs.

Normally in multiple floors (up/down) scenarios there are two vacuums here is the same vacuum with "multiple maps data". Do I'm correct?

Then you will have two cards one for the lower floor and one for upper floor as I it is possible to dynamically change the segments data in the card this implies you each one cards with the new segments data. Of course both cards will show the same images, but for instance the one configured for the lower floor will have the correct segment data (you may do reload the camera if you want it to generate the new segments).

To use the same card, same vacuum, multiple floors. Mmmm.. it's quite an interesting scenario.. but honestly I don't like the idea to mess around with this for now.. please understand..

I think I got finally what we were discussing earlier.. it will probably take a little but I think we can probably found a solution will update you soon.

**Comment by carefulcomputer** [comment]
This is brilliant!! I like the idea of swapping out the card based on floor. The maploader also sets the floor name as a 'select' sensor in home assistant. So based on that sensor, I can swap out the card.  I will give it a try. And thanks for creating this amazing card !

**Reply by sca075** [reply]
Well I didn't create the card but the camera. But I know that the card as I wrote support different floors (will need to confirm with Piotr later on how it works)
So @carefulcomputer as I wrote please hold on and will come back to you as soon I can.. probably I will get in contact also with the creator of this plugin you use and if I understand the technical part.. well it should not be that difficult ;)

**Reply by sca075** [reply]
@carefulcomputer did you have a look at this? https://github.com/PiotrMachowski/lovelace-xiaomi-vacuum-map-card/issues/248 

It looks it could help to solve the issue by using a service call and change the map. 

I'm still aboard and unfortunately I couldn't have a look personally on how to implement this functionality but it looks there's the possibility to have to cards configured at the same time in the same output.
